### ISSUE TITLE
[Distribution Build] Integ-Test Failure RestDeleteTransformActionIT

### ISSUE THREAD

**Issue by peternied** [opensearch-project/index-management#197]
While running tests on, OpenSearch-Build @805fbec, ARM machine, `./test.sh integ-test https://ci.opensearch.org/ci/dbc/bundle-build/1.2.0/1032/linux/arm64` found this failure:
```
REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.indexmanagement.transform.resthandler.RestDeleteTransformActionIT.test deleting a transform that doesn't exist and config index doesn't exist" -Dtests.seed=FA5E1A76E2C442B7 -Dtests.security.manager=false -Dtests.locale=ms -Dtests.timezone=Europe/Mariehamn -Druntime.java=14
NOTE: leaving temporary files on disk at: /tmp/tmpf4f7ne2z/index-management/build/testrun/integTest/temp/org.opensearch.indexmanagement.transform.resthandler.RestDeleteTransformActionIT_FA5E1A76E2C442B7-001
NOTE: test params are: codec=Asserting(Lucene87): {}, docValues:{}, maxPointsInLeafNode=328, maxMBSortInHeap=7.653077620742154, sim=Asserting(RandomSimilarity(queryNorm=false): {}), locale=ms, timezone=Europe/Mariehamn
NOTE: Linux 5.4.147-83.259.amzn2int.aarch64 aarch64/AdoptOpenJDK 14.0.2 (64-bit)/cpus=32,threads=1,free=468806488,total=536870912
NOTE: All tests run in this JVM: [IndexManagementIndicesIT, RestPreviewTransformActionIT, RestDeleteTransformActionIT]
```

#### test deleting a transform that doesn't exist and config index doesn't exist

```
java.lang.AssertionError: expected:<BAD_REQUEST> but was:<NOT_FOUND>
	at __randomizedtesting.SeedInfo.seed([FA5E1A76E2C442B7:41480812FE7373CF]:0)
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at org.opensearch.indexmanagement.transform.resthandler.RestDeleteTransformActionIT.test deleting a transform that doesn't exist and config index doesn't exist(RestDeleteTransformActionIT.kt:88)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.base/java.lang.Thread.run(Thread.java:832)
```

This issue needs to be fixed for the 1.2.0 release.

**Comment by downsrob** [comment]
I can consistently replicate this locally when I run the test individually, no matter what the test seed.
When I run the test suite, I have only been able to replicate this when I use the provided test seed. This does not seem to be a flakey test failure.

**Comment by downsrob** [comment]
This is a test failure caused by an implicit dependency on the ism index existing at the start of the test. The test attempts to delete the index, and this is throwing the unexpected NOT_FOUND error. Going to fix the test in a PR now.

**Comment by downsrob** [comment]
This issue is fixed on main, but I haven't marked it as closed yet as we are in the process of backporting to 1.2

**Comment by downsrob** [comment]
Fix merged to 1.2
